pkg/ingester/ingester_ring.go

```diff
@@ -94,6 +101,10 @@ func (cfg *RingConfig) RegisterFlags(f *flag.FlagSet, logger log.Logger) {
 	flagext.DeprecatedFlag(f, prefix+"join-after", "Deprecated: this setting was used to set a period of time to wait before joining the hash ring. Mimir now behaves as this setting is always set to 0s.", logger)
 	f.DurationVar(&cfg.MinReadyDuration, prefix+"min-ready-duration", 15*time.Second, "Minimum duration to wait after the internal readiness checks have passed but before succeeding the readiness endpoint. This is used to slowdown deployment controllers (eg. Kubernetes) after an instance is ready and before they proceed with a rolling update, to give the rest of the cluster instances enough time to receive ring updates.")
 	f.DurationVar(&cfg.FinalSleep, prefix+"final-sleep", 0, "Duration to sleep for before exiting, to ensure metrics are scraped.")
+
+	// TokenGenerator
+	f.StringVar(&cfg.TokenGeneratorStrategy, prefix+tokenGenerationStrategyFlag, randomTokenGeneration, "Specifies the strategy used for generating tokens for ingesters. Possible values are \""+randomTokenGeneration+"\" (default) and \""+spreadMinimizingTokenGeneration+"\"")
```

Comment:
[nit] No need to specify the default here (it's already injected when generating the config doc. Also can simplify with `strings.Join()`.

```suggestion
	f.StringVar(&cfg.TokenGeneratorStrategy, prefix+tokenGenerationStrategyFlag, randomTokenGeneration, fmt.Sprintf("Specifies the strategy used for generating tokens for ingesters. Supported values are: %s.", strings.Join([]string{randomTokenGeneration, spreadMinimizingTokenGeneration}, ",")))
```